### DISCUSSION TITLE
Problem with modifying DOI field in BibTeX file

### DISCUSSION THREAD

**Discussion by MikeChanV8** [retorquere/zotero-better-bibtex#2426] [Q&A] [answered]
I am writing to report an issue with modifying the DOI field in a export bibtex file. Specifically, I am trying to replace the existing DOI field with a new note field that includes a hyperlink to the DOI URL. However, the resulting output is not what I expected.

Here's the original DOI field that I want to modify:

doi={10.1049/iet-pel.2016.0483};

And here's the code in postscript that I am using to replace it with a new note field:

```
if (Translator.BetterTeX) {
    if (tex.has['doi']) {
        const doiValue = tex.has['doi'].value;
        const doiUrl = `https://doi.org/${doiValue}`;
        noteValue =`doi:\href{${doiUrl}}{${doiValue}}`;
        tex.add({
            name: 'note',
            value:  noteValue
        });
        delete tex.has['doi'];
    }
}
```

However, the resulting output that I get is not what I expected. That is
```
note = {doi:href\{https://doi.org/10.1049/iet-pel.2016.0483\}\{10.1049/iet-pel.2016.0483\}}
```
The expected output should be:
```
note = {doi:\href{https://doi.org/10.1049/iet-pel.2016.0483}{10.1049/iet-pel.2016.0483}}
```
Specifically, the character '' appears before the opening and closing braces of the hyperlink, and the character '' before 'href' is lost.  This is not what I expected, and I believe there might be a problem with the code that I am using.

I would appreciate any help or suggestions on how to fix this issue.

Thank you for your attention and assistance.

**Comment by retorquere** [comment, marked as answer]
```javascript
if (Translator.BetterTeX) {
  if (tex.has.doi) {
    tex.add({ name: 'note', value: `doi:<a href="https://doi.org/${tex.has.doi.value}">${tex.has.doi.value}</a>`, html: true })
    delete tex.has.doi
  }
}
```

**Reply by MikeChanV8** [reply]
It works! Thank you so much for your help!

**Reply by MikeChanV8** [reply]
In the newest version V6.7.55, postscript is completely disabled.

**Reply by retorquere** [reply]
It isn't disabled at all, which I know because my test suite runs several tests which have postscripts. For diagnosis of the problem you are experiencing, please open a new issue and send a debug log.
